src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -205,20 +205,14 @@ static void keepalive_watchdog_fired_locked(
 static void maybe_reset_keepalive_ping_timer_locked(grpc_chttp2_transport* t);
 
 namespace {
-void MaybeRecordTransportAnnotation(grpc_chttp2_stream* s,
-                                    absl::string_view annotation) {
-  GPR_ASSERT(s->context);
-  if (!grpc_core::IsTraceRecordCallopsEnabled()) {
-    return;
+grpc_core::CallTracerInterface* CallTracerIfEnabled(grpc_chttp2_stream* s) {
+  if (!s->context || !grpc_core::IsTraceRecordCallopsEnabled()) {
```

Comment:
nit: prefer explicit checks against nullptr, such as `if (s->context == nullptr || ...)`.

Reply:
Done. I changed the tracer check below to compare with nullptr too. AFAICT grpc supports c++ versions before 17 so I didn't use
if (auto* tracer = ....; tracer != nullptr)

If that's not the case let me know since that seems cleaner if allowed.

Reply:
> grpc supports c++ versions before 17

Indeed, gRPC still needs to support c++14. This looks good, thanks.